feature-libs/asm/components/customer-emulation/customer-emulation.component.ts

```diff
@@ -1,20 +1,43 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import { User, UserService } from '@spartacus/core';
+import {
```

Comment:
I think the cart binding feature should be in its own component.  Adding all this logic to the existing customer emulation components gives too much responsibility to that component.  Smaller more focused components are better to make things easier to maintain and evolve.  It also makes the unit tests more manageable.